webapp/channels/src/components/user_settings/modal/user_settings_modal.tsx

```diff
@@ -84,6 +98,22 @@ class UserSettingsModal extends React.PureComponent<Props, State> {
 
     componentDidMount() {
         document.addEventListener('keydown', this.handleKeyDown);
+
+        if (this.props.adminMode && this.props.userID) {
+            this.setState({loading: true});
```

Comment:
Could this state be set in the constructor?

Reply:
Setting loading to true in constructor causes some edge cases, thats why I moved it from constructor to here.

Reply:
I'm surprised that that doesn't cause the modal to briefly render without a loader. I guess that doesn't happen because the render method looks at `userID` though